### PR TITLE
Clang linker workaround 

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes

1. Workaround clang linker bug for offload using object target. The code compiles now but not fully running yet due other bugs.
2. Split qmcwfs target into qmcwfs and qmcwfs_cuda to facilicate 1.
3. introduce spir64 for intel next gen compiler.

## What type(s) of changes does this code introduce?
- Build related changes

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Ryzen-box

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by markdewing** [comment]
Item #3 should be a separate PR.  It looks like the files are completely disjoint from the files in the other two points, so it shouldn't be that hard to split out.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -243,14 +243,14 @@ ENDIF()
...
-  ENDIF(OHMMS_DIM  MATCHES 3)
+  ENDIF(OHMMS_DIM MATCHES 3)
 
   #Need to add this if nothing else but miniapps is compiled
-  if(QMC_BUILD_SANDBOX_ONLY)
-    set(PARTICLE  ${PARTICLE} QMCWaveFunctions/WaveFunctionComponent.cpp)
-  endif()
+  IF(QMC_BUILD_SANDBOX_ONLY)
+    SET(PARTICLE ${PARTICLE} QMCWaveFunctions/WaveFunctionComponent.cpp)
+  ENDIF(QMC_BUILD_SANDBOX_ONLY)
```
Comment (markdewing) [inline src/CMakeLists.txt:253]: Adding the condition in 'endif' is for backwards compatibility (according to the 'endif' docs).     It probably shouldn't be added when cleaning up CMake files.

Comment (ye-luo) [reply src/CMakeLists.txt:253]: I think we can be flexible here. Sometimes putting more details in endif helps reading.

**Comment by ye-luo** [comment]
> Initial comments without looking at the code:
> 
>     1. I agree with Mark, please split.
> 
>     2. Please specify which clang you are referring to. Does IBM XL on Summit work (main concern)? 9.x or 10.0?

1. I removed the housekeeping change. 
2. IBM is not affected. Now I'm able to develop code without the slow XL. Clang develop is what I need. 9 and 10 are outdated to me.

**Comment by prckent** [comment]
It is good that you will be able to develop faster with this change. I would still like to know which compiler versions build QMCPACK with this change. Will release clang10.0 with the openmp target backend compile QMCPACK successfully? or does this require a change made after the 10.0 release branch? Does AOMP still work? Nightlies will be added using these compiler shortly.

**Comment by ye-luo** [comment]
XL and AOMP always run. The fix for math header lands after Clang 10 unfortunately.

**Comment by prckent** [comment]
Merging ahead of nightlies since CI is temporarily offline.

**Pull request merged**